### PR TITLE
Fast jar performance improvements

### PR DESCRIPTION
**Pull request by stuartwdouglas**

### CODE REVIEW

File: independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/SerializedApplication.java [hunk 1]
```diff
@@ -152,6 +168,24 @@ private static void writeJar(DataOutputStream out, Path jar) throws IOException
...
 
+    private static void collectPackages(Path jar, Set<String> dirs) throws IOException {
+        try (JarFile zip = new JarFile(jar.toFile())) {
+            Enumeration<? extends ZipEntry> entries = zip.entries();
+            while (entries.hasMoreElements()) {
+                ZipEntry entry = entries.nextElement();
+                if (!entry.isDirectory()) {
+                    //some jars don't have correct directory entries
+                    //so we look at the file paths instead
+                    //looking at you h2
```
Comment (geoand) [inline independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/SerializedApplication.java:179]: How weird...

**Comment by geoand** [comment]
Don't we want the `findClass` method to be defined (instead of "inlining" it into `loadClass`)? Can't letting it default to throwing `ClassNotFoundException` cause problems?

**Comment by stuartwdouglas** [comment]
findClass should never be called, as it is usually called by loadClass which we have overriden.

**Comment by geoand** [comment]
> findClass should never be called, as it is usually called by loadClass which we have overriden.

Yeah, I'm just asking if there is any chance based on your experience that libraries choose to call `findClass` instead.

**Comment by stuartwdouglas** [comment]
Not if they expect things to work correctly, it's not part of the public API.

**Comment by geoand** [comment]
OK, cool

**Comment by stuartwdouglas** [comment]
This should be good to go now.

**Review by geoand — APPROVED** [review]
Great!

**Pull request merged**